DependencyInjection/Configuration.php

```diff
@@ -81,13 +81,25 @@ public function getConfigTreeBuilder()
                                         ->ifTrue(function ($v) { isset($v['route_basepath']) && isset($v['route_basepaths']); })
                                         ->thenInvalid('Found values for both "route_basepath" and "route_basepaths", use "route_basepaths" instead.')
                                     ->end()
+                                    ->beforeNormalization()
+                                        ->ifTrue(function ($v) { return isset($v['route_basepath']); })
+                                        ->then(function ($v) {
+                                            @trigger_error('The route_basepath setting is deprecated as of version 1.4 and will be removed in 2.0. Use route_basepaths instead.', E_USER_DEPRECATED);
```

Comment:
The deprecation handling system in Symfony is now good enough to immediately trigger deprecation notices. As the `@` is used, these errors will be silent by default (unless someone uses a custom error handler, which is only the case when you use the PHPunit Bridge or the deprecation catcher from the web dev toolbar, in both cases one is looking for reports about deprecated feature usages).